### DISCUSSION TITLE
Why docs suggest installing Tailwind as dev dependency?

### DISCUSSION THREAD

**Discussion by bwmirek** [tailwindlabs/tailwindcss#8220] [General]
Hi 👋
Should Tailwind really be installed as dev dependency? In the end, it is necessary to have it during the production build. I've had a problem with such approach a few times when my system administrator set up an automatic build with production flag and the system crashed because of Tailwind marked as dev dependency. So what's the logic behind that?

**Comment by navaneeth-dev** [comment]
Bump anyone?

**Reply by bwmirek** [reply]
I think I understood this logic a bit. Being in dev dependencies means that this very specific library is meant to be used during development **and** building phase. It is not served to end user as it is (frontend case). You're using Tailwind to **build** an actual CSS.

Personally, I find it a bit clumsy. In my case, I can't install dependencies and run build with `--production` flag.

**Reply by phil294** [reply]
Your first paragraph is correct. That's why tailwind suggesting itself to be installed as a dev dep is the correct choice.

Regarding your builds: If you tell npm to install your dependencies in production, so `NODE_ENV=production npm install`, you're telling it that you *already are* in production mode. It does *not* mean that you're trying to install the dependencies required *for building*. Building = dev, running = production (usually). You would usually only do a production install if you're running a node server, which in turn may or may not have undergone some preprocessing. It's double confusing though because if you do `NODE_ENV=production npm build`, in this case you *are* usually signifying to your build tool your *target* output, not the current one, so kinda the opposite.

**Comment by sacru2red** [comment]
Hmm... isn't it common to not run builds on production machines?
